test/jdk/javax/swing/text/html/TestResetSelectForm.java

```diff
@@ -0,0 +1,153 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.util.Arrays;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JList;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8295248
+ * @summary Verify select items of HTML form are cleared by reset
+ */
+public class TestResetSelectForm {
+    private Robot robot;
+    private JEditorPane editor;
+    private JFrame frame;
+
+    private int listHeight;
+    private Point loc;
+    private Color baseColor;
+
+    private final int width = 200;
+    private final int height = 200;
+    private final int LIMIT = 10;
+    private final String[] numbers = {"1","2","3","4","5"};
+
+    private int getListHeight() {
+        JList<String> dummy = new JList<>(numbers);
+        return dummy.getPreferredSize().height;
+    }
+
+    private void setup() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("<html><body><form action=\"http://localhost\">" +
+                  "<select name=select id=\"mySelect\" size=\"" +
+                  numbers.length + "\" multiple>");
+        Arrays.stream(numbers).forEach(s->{sb.append(
+                  "<option>" + s + "</option>");});
+        sb.append("</select><br>" +
+                  "<input type=reset name=reset value=\"reset\"/>" +
+                  "<br>" +
+                  "<input type=submit name=submit value=\"submit\"/>" +
+                  "</form></body></html>");
+        editor = new JEditorPane("text/html", sb.toString());
+        frame = new JFrame();
+        frame.setSize(width, height);
+        frame.setLayout(new BorderLayout());
+        frame.add(editor, BorderLayout.CENTER);
+        frame.setVisible(true);
+    }
+
+    private static int getMaxColorDiff(Color c1, Color c2) {
+        return Math.max(Math.abs(c1.getRed() - c2.getRed()),
+                 Math.max(Math.abs(c1.getGreen() - c2.getGreen()),
+                          Math.abs(c1.getBlue() - c2.getBlue())));
+    }
+
+    private void checkNoListSelection() {
+        int len = numbers.length;
+        for (int i = 0; i < len; i++) {
+            int pos = (int)(listHeight * (i + 0.5) / len);
+            Color c = robot.getPixelColor(loc.x + 30, loc.y + pos);
+            if (getMaxColorDiff(baseColor, c) > LIMIT) {
+                throw new RuntimeException("Unexpected List selection at " +
+                    (i+1) + ", " + baseColor + "," + c);
```

Comment:
The points mentioned seem valid since I see the same things when I test it myself. OptionListModel's clearSelection is being called without the drawing updating. The fix seems logical and works as intended.

Just to nitpick, adding a space before and after the '+' in '(i+1)' here would match the pattern of all the other additions.

Reply:
@DamonGuy Thank you for the comments. Fixed.